### DISCUSSION TITLE
Is there a way to enable inline type inference.

### DISCUSSION THREAD

**Discussion by laycookie** [mrcjkb/rustaceanvim#346] [Q&A] [answered]
Hello Im switching from JetBrains rust rover to neovim and Im really missing the type inference from there. Does this plugin provides the following functionality or should I get some other plugin?

**Comment by mrcjkb** [comment, marked as answer]
Hey 👋 

See https://github.com/mrcjkb/rustaceanvim/discussions/46

**Reply by laycookie** [reply]
@mrcjkb Thank you, I had seen this issue mentioned in #122 but I was thinking that it was talking about the errors. I know this is getting a little bit off topic now but what about the errors showing not showing up? I have the text underlined e.t.c. but I dont get the reason behind that error like in rust_tools can you explain how to enable that because it dosent show up after setting `vim.lsp.inlay_hint.enable` to true.

**Reply by mrcjkb** [reply]
Are you talking about diagnostics?
That could have multiple reasons. 

Here are the main two differences between this plugin and rust-tools:

- Some neovim distributions configure lspconfig (and by extension rust-tools) to use mason.nvim.
  Because this plugin doesn't depend on lspconfig, that doesn't work for this plugin (and I don't recommend using mason for rust-analyzer, see `:h rustaceanvim.mason`).
- If you have clippy installed, rustaceanvim will configure rust-tools to enable clippy for diagnostics. You can try disabling that (see `:h rustaceanvim.config`).

Apart from that, I recommend following the troubleshooting guide in the readme.
